allennlp/modules/transformer/__init__.py

```diff
@@ -8,147 +8,117 @@
-
-class TinyTransformer(Model, TransformerModule):
-    _huggingface_mapping = TransformerEncoder._huggingface_mapping
-    def __init__(self,
-                 vocab,
-                 embedding_file,
-                 embedding_dim: int,
-                 encoder: TransformerEncoder,
-                ):
-        super().__init__(vocab)
-        self.embedding_layer = Embedding(pretrained_file=embedding_file,
-                                         embedding_dim=embedding_dim,
-                                         projection_dim=encoder._hidden_size,
-                                         vocab=vocab)
-
-        self.encoder = encoder
-
-    def forward(self, inp):
-        embedded = self.embedding_layer(inp)
-        outputs = self.encoder(embedded)
-        return outputs
-
-encoder = TransformerEncoder(num_hidden_layers=4,
-                             hidden_size=80,
-                             intermediate_size=40,
-                             num_attention_heads=8,
-                             attention_dropout=0.1,
-                             hidden_dropout=0.1,
-                             activation="relu")
-
-tiny = TinyTransformer(vocab, embedding_file, embedding_dim=300, encoder=encoder)
+embedding_file = str(self.FIXTURES_ROOT / "embeddings/glove.6B.300d.sample.txt.gz")
+
+class TinyTransformer(TokenEmbedder):
+    def __init__(self, vocab, embedding_dim, hidden_size, intermediate_size):
+        super().__init__()
+        self.embeddings = Embedding(
+            pretrained_file=embedding_file,
+            embedding_dim=embedding_dim,
+            projection_dim=hidden_size,
+            vocab=vocab,
+        )
+
+        self.transformer = TransformerBlock(
+            num_hidden_layers=4,
+            hidden_size=hidden_size,
+            intermediate_size=intermediate_size,
+        )
+
+    @overrides
+    def forward(self, token_ids: torch.LongTensor):
+        x = self.embeddings(token_ids)
+        x = self.transformer(x)
+        return x
+
+tiny = TinyTransformer(self.vocab, embedding_dim=300, hidden_size=80, intermediate_size=40)
 ```
 
 2. Use the first 4 layers of `bert-base-uncased`.
 
 ```
-from transformers.modeling_auto import AutoModel
-pretrained = AutoModel.from_pretrained('bert-base-uncased')
-encoder = TransformerEncoder.from_pretrained_module(pretrained.encoder, num_hidden_layers=4)
-
-tiny = TinyTransformer(vocab, embedding_file, embedding_dim=300, encoder=encoder)
+pretrained = cached_transformers.get("bert-base-uncased", False)
+
+class SmallTransformer(TokenEmbedder):
+    def __init__(self):
+        super().__init__()
+        self.embeddings = TransformerEmbeddings.from_pretrained_module(pretrained)
+
+        self.transformer = TransformerBlock.from_pretrained_module(
+            pretrained, num_hidden_layers=4
+        )
+
+    @overrides
+    def forward(self, token_ids: torch.LongTensor):
+        x = self.embeddings(token_ids)
+        x = self.transformer(x)
+        return x
```

Comment:
I don't think this is completely correct, because it neglects token type IDs. That's not a thing to fix in this PR though.